libs/dh/message-archive/feature-search/src/lib/filters.component.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  DocumentType,
+  GetArchivedMessagesQueryVariables,
+} from '@energinet-datahub/dh/shared/domain/graphql';
+
+@Component({
+  selector: 'dh-message-archive-search-filters',
+  standalone: true,
+  imports: [
+    ReactiveFormsModule,
+    TranslocoDirective,
+    VaterSpacerComponent,
+    VaterStackComponent,
+    WattButtonComponent,
+    WattDateChipComponent,
+    WattDatetimepickerComponent,
+    WattDropdownComponent,
+    WattFilterChipComponent,
+    WattFormChipDirective,
+    WattModalActionsComponent,
+    WattModalComponent,
+    DhDropdownTranslatorDirective,
+  ],
+  template: `
+    <form
+      vater-stack
+      direction="row"
+      gap="s"
+      tabindex="-1"
+      [formGroup]="form.root"
+      *transloco="let t; read: 'messageArchive.filters'"
+    >
+      @if (isSearchingById()) {
+        <watt-filter-chip [formControl]="this.form.controls.includeRelated">
+          {{ t('includeRelated') }}
+        </watt-filter-chip>
+      } @else {
+        <watt-dropdown
+          [formControl]="form.controls.documentTypes"
+          [chipMode]="true"
+          [multiple]="true"
+          [options]="form.documentTypeOptions"
+          [placeholder]="t('documentType')"
+          [getCustomTrigger]="getDocumentTypeTrigger"
+          dhDropdownTranslator
+          translateKey="messageArchive.documentType"
+        />
+
+        <watt-dropdown
+          [formControl]="form.controls.businessReasons"
+          [chipMode]="true"
+          [multiple]="true"
+          [options]="form.businessReasonOptions"
+          [placeholder]="t('businessReason')"
+          [getCustomTrigger]="getBusinessReasonTrigger"
+          dhDropdownTranslator
+          translateKey="messageArchive.businessReason"
+        />
+
+        @if (form.isActorControlsEnabled()) {
+          <watt-dropdown
+            [formControl]="form.controls.senderId"
+            [chipMode]="true"
+            [options]="form.actorOptions()"
+            [placeholder]="t('sender')"
+          />
+
+          <watt-dropdown
+            [formControl]="form.controls.receiverId"
+            [chipMode]="true"
+            [options]="form.actorOptions()"
+            [placeholder]="t('receiver')"
+          />
+        }
+
+        <watt-date-chip [formControl]="form.controls.start" [placeholder]="t('start')" />
+
+        <watt-date-chip [formControl]="form.controls.end" [placeholder]="t('end')" />
+      }
+
+      <vater-spacer />
+
+      <watt-button variant="text" icon="undo" (click)="reset()">{{ t('reset') }}</watt-button>
+    </form>
+  `,
+})
+export class DhMessageArchiveSearchFiltersComponent {
+  isSearchingById = input(false);
+  filter = output<GetArchivedMessagesQueryVariables>();
+  clear = output();
+
+  form = inject(DhMessageArchiveSearchFormService);
+
+  getDocumentTypeTrigger = (value: string | string[]) =>
+    this.form.getDocumentTypeIdentifier(value[0] as DocumentType);
```

Comment:
Issue, if this is a string getDocumentTypeIdentifier will be set to the first letter of the string?

Reply:
True, but that is only if the dropdown is not in `multiple` selection mode, which it always is. So this type is just to satisfy the compiler, `value` will always be an array. But I agree that it looks like a bug - I will add a comment.

Reply:
bfc2ba026497dbe4cd8936fb76419f47b0b13169